packages/core/src/ctype/CType.spec.ts

```diff
@@ -28,76 +27,52 @@ const encodedAliceDid = ApiMocks.mockChainQueryReturn(
 )
 const didAlice = 'did:kilt:4p6K4tpdZtY3rNqM2uorQmsS6d3woxtnWMHjtzGftHmDb41N'
 
-describe('CType', () => {
-  let claimCtype: ICType
-  let claimContents: any
-  let claim: IClaim
-  beforeAll(async () => {
-    claimCtype = CType.fromProperties('CtypeModel 2', {
-      name: { type: 'string' },
-    })
-
-    claimContents = {
-      name: 'Bob',
-    }
-
-    claim = Claim.fromCTypeAndClaimContents(claimCtype, claimContents, didAlice)
+it('consistent CType id generation', () => {
+  const ctypeV1 = CType.fromProperties('CtypeModel 1', {
+    'first-property': { type: 'integer' },
+    'second-property': { type: 'string' },
   })
 
-  it('makes ctype object from schema without id', () => {
-    const ctype = CType.fromProperties('CtypeModel 1', {
+  expect(ctypeV1.$id).toMatchInlineSnapshot(
+    `"kilt:ctype:0x12c8edb42b455aa6c29fabda8f3768bd1e8577f0618f122072828e41b6f4f728"`
+  )
+
+  const ctypeV0 = CType.fromProperties(
+    'CtypeModel 1',
+    {
       'first-property': { type: 'integer' },
       'second-property': { type: 'string' },
-    })
-
-    expect(ctype.$id).toBe(
-      'kilt:ctype:0xba15bf4960766b0a6ad7613aa3338edce95df6b22ed29dd72f6e72d740829b84'
-    )
-  })
+    },
+    'draft-01'
+  )
 
-  it('verifies the claim structure', () => {
-    expect(() =>
-      CType.verifyClaimAgainstSchema(claim.contents, claimCtype)
-    ).not.toThrow()
-    claim.contents.name = 123
-    expect(() =>
-      CType.verifyClaimAgainstSchema(claim.contents, claimCtype)
-    ).toThrow()
-  })
-
-  it('throws error on faulty input', () => {
-    const wrongHashCtype: ICType = {
-      ...claimCtype,
-      $id: 'kilt:ctype:0x1234',
-    }
-    const faultySchemaCtype: ICType = {
-      ...claimCtype,
-      properties: null as unknown as ICType['properties'],
-    }
+  expect(ctypeV0.$id).toMatchInlineSnapshot(
+    `"kilt:ctype:0xba15bf4960766b0a6ad7613aa3338edce95df6b22ed29dd72f6e72d740829b84"`
```

Comment:
Isn't this supposed to be the same as the old value, which is `kilt:ctype:0xdfe0fd007b86ef280f366d25547873da0305538d8190a2eae2f2d875cb5cf30d`?

Reply:
That's impossible unfortunately as the id is the Schema hash and the Schema hash is computed over the entire Schema (excluding $id obviously) so including the meta Schema reference $schema and the additionalProperties assertion

Reply:
In short: new version, new ctype.

Reply:
Got it. There's something new even for old CTypes.